src/transactions/TransactionFrame.cpp

```diff
@@ -1544,19 +1386,20 @@ TransactionFrame::removeAccountSigner(AbstractLedgerTxn& ltxOuter,
     }
 }
 
-bool
+std::pair<bool, MutableTxResultPtr>
```

Comment:
Do we actually need a bool here? The only way this succeeds is by returning a non-null result containing success code, right? So the callers could just look for that.

Reply:
I think this would make the interface more messy. For example, the caller would have to know if the underlying type of the TransactionFrame to check for `txFEE_BUMP_INNER_SUCCESS` vs `txSUCCESS`. It's also just more boiler plate code for the callers, since most every caller of `checkValid` will need to check if the TX succeeded or not, so the logic will need to be reimplemented after every call.

Reply:
The boiler plate issue can be nicely (and more readably) solved by providing `virtual bool isSuccess()` method on the result. That's IMO more expressive/readable then parsing the pair and especially better than `.first`.

Reply:
Done